R/run_episode_file.R

```diff
@@ -326,3 +326,76 @@ join_cohort_lookups <- function(ep_file_data, year) {
 
   return(join_cohort_lookups)
 }
+
+
+#' Join SPARRA and HHG
+#'
+#' @param ep_file_data Episode file data.
+#' @param year financial year, e.g. '1920'
+#'
+#' @return The data including the SPARRA and HHG variables matched
+#' on to the episode file.
+#'
+join_sparra_hhg <- function(ep_file_data, year) {
+  fy <- year
+  next_fy <- as.character(glue::glue("{substr(as.numeric(fy), 3, 4)}{substr(as.numeric(fy)+1, 3, 4)}"))
+
+  if (!check_year_valid(year, "Sparra-HHG")) {
+    join_sparra_hhg_file <- ep_file_data %>%
+      dplyr::full_join(
+        read_file(get_sparra_path(fy)) %>%
+          dplyr::rename(
+            chi = "upi_number",
+            sparra_start_fy = "sparra_risk_score"
+          ),
+        by = "chi"
+      ) %>%
+      dplyr::full_join(
+        read_file(get_hhg_path(fy)) %>%
+          dplyr::rename(
+            chi = "upi_number",
+            hhg_start_fy = "hhg_score"
+          ),
+        by = "chi"
+      )
```

Comment:
Can you make the next_fy bit into a function, please?
```suggestion
  fy <- as.integer(substr(year, 1, 2))
  next_fy <- paste0(fy + 1L, fy + 2L)

  if (!check_year_valid(year, "Sparra-HHG")) {
    join_sparra_hhg_file <- ep_file_data %>%
      dplyr::full_join(
        read_file(get_sparra_path(year)) %>%
          dplyr::rename(
            chi = "upi_number",
            sparra_start_fy = "sparra_risk_score"
          ),
        by = "chi"
      ) %>%
      dplyr::full_join(
        read_file(get_hhg_path(year)) %>%
          dplyr::rename(
            chi = "upi_number",
            hhg_start_fy = "hhg_score"
          ),
        by = "chi"
      )
```